src/styles/getReportActionContextMenuItemStyles.js

```diff
@@ -0,0 +1,107 @@
+import CONST from '../CONST';
+import themeColors from './themes/default';
+import styles from './styles';
+import variables from './variables';
+import fontFamily from './fontFamily';
+
+/**
+ * Generate a style for the background color of the button, based on its current state.
+ *
+ * @param {String} [buttonState] - One of {'default', 'hovered', 'pressed'}
+ * @returns {Object}
+ */
+function getButtonBackgroundColorStyle(buttonState = CONST.BUTTON_STATES.DEFAULT) {
+    switch (buttonState) {
+        case CONST.BUTTON_STATES.HOVERED:
+            return {backgroundColor: themeColors.hoverComponentBG};
+        case CONST.BUTTON_STATES.PRESSED:
+            return {backgroundColor: themeColors.activeComponentBG};
+        case CONST.BUTTON_STATES.DEFAULT:
+        default:
+            return {};
+    }
+}
+
+/**
+ * Generate styles for the buttons in the mini ReportActionContextMenu.
+ *
+ * @param {String} [buttonState] - One of {'default', 'hovered', 'pressed'}
+ * @returns {Array}
+ */
+function getMiniButtonStyle(buttonState = CONST.BUTTON_STATES.DEFAULT) {
+    return [
+        getButtonBackgroundColorStyle(buttonState),
+        styles.p1,
+        styles.mv1,
+        styles.mh1,
+        {borderRadius: variables.componentBorderRadiusSmall},
+    ];
+}
+
+/**
+ * Generate styles for the rows in the larger ReportActionContextMenu.
+ *
+ * @param {String} [buttonState] - One of {'default', 'hovered', 'pressed'}
+ * @returns {Array}
+ */
+function getBigButtonStyle(buttonState = CONST.BUTTON_STATES.DEFAULT) {
+    return [
+        getButtonBackgroundColorStyle(buttonState),
+        styles.flex1,
+        styles.flexRow,
+        styles.alignItemsCenter,
+        styles.p3,
+    ];
+}
+
+/**
+ * Get the fill color for the icons in the menu depending on the state of the button they're in.
+ *
+ * @param {String} [buttonState] - One of {'default', 'hovered', 'pressed'}
+ * @returns {String}
+ */
+function getIconFillColor(buttonState = CONST.BUTTON_STATES.DEFAULT) {
+    switch (buttonState) {
+        case CONST.BUTTON_STATES.HOVERED:
+            return themeColors.text;
+        case CONST.BUTTON_STATES.PRESSED:
+            return themeColors.heading;
+        case CONST.BUTTON_STATES.DEFAULT:
+        default:
+            return themeColors.icon;
+    }
+}
+
+/**
+ * Get the style of the text in the ReportActionContextMenu, depending on the state of the button it's it.
+ *
+ * @param {String} [buttonState] - One of {'default', 'hovered', 'pressed'}
+ * @returns {Array}
+ */
+function getTextStyle(buttonState = CONST.BUTTON_STATES.DEFAULT) {
```

Comment:
What is the benefit of doing styles this way instead of just adding the style into style.js like we do with everything else?

Reply:
These are dynamic styles: i.e they're different depending on the state of the button.

>  just adding the style into style.js like we do with everything else?

This is not entirely correct - there are several other places where we've generated dynamic styles like this:

- https://github.com/Expensify/Expensify.cash/blob/master/src/styles/getModalStyles.js
- https://github.com/Expensify/Expensify.cash/blob/master/src/styles/getTooltipStyles.js
- https://github.com/Expensify/Expensify.cash/blob/380a148fdd47ce4c37e07ce647e1d9d31af8eb88/src/styles/styles.js#L1027-L1069